pallets/vanchor/src/lib.rs

```diff
@@ -114,6 +115,9 @@ pub mod pallet {
 		type LinkableTree: LinkableTreeInterface<pallet_linkable_tree::LinkableTreeConfigration<Self, I>>
 			+ LinkableTreeInspector<pallet_linkable_tree::LinkableTreeConfigration<Self, I>>;
 
+		/// The key storage type
+		type KeyStorage: KeyStorageInterface<pallet_key_storage::KeyStorageConfiguration<Self, I>>;
```

Comment:
I would consider generalizing this simply over `Self::AcocuntId` instead of the `KeyStorageConfiguration` and then having the `pallet_key_storage` implement `KeyStorageInterface`. Then we hook it up in the runtime instead of importing this pallet into the `pallet-vanchor`.

Reply:
`pallet_key_storage` actually implements `KeyStorageInterface` What do you mean hook it up in the runtime? You mean making use of the pallet hooks?

Reply:
You shouldn't need to import `pallet-key-storage` to import the trait that exposes the functionality.